src/components/Details.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+const Details = ({ details, setDetails }) => {
+  let numOfPurch = details.purchaseDates.length;
+  const lastPurch = numOfPurch
+    ? Math.max(...details.purchaseDates)
+    : details.addedDate;
+  let futurePurch = lastPurch + details.frequency;
```

Comment:
I love how readable all of this code is! Everything is clearly laid out, and all variables names clearly clue anyone reading the code.

I just checked Safari, Chrome, and Firefox and each browser, or maybe Firestore, is guaranteeing insert order for the `purchaseDates` array. I think we can remove the call to `Math.max` because the `purchaseDates` array will always be sorted and we just need to grab the last element (the most recent purchase) in the `purchaseDates` array.

Reply:
I used `Math.max` to keep things consistent with other components using the same value. Perhaps we should have just stashed last purchase date as a separate variable in Firebase to avoid this!